frontend/src/features/SideWindow/MissionForm/utils.ts

```diff
@@ -0,0 +1,68 @@
+import { omit } from 'ramda'
+
+import { Mission, MissionData, MissionSource } from '../../../domain/types/mission'
+import { FormError } from '../../../libs/FormError'
+import { hasMissingOrUndefinedValues } from '../../../utils/hasMissingOrUndefinedValues'
+
+import type { MissionFormValues } from './MainForm/types'
+
+/**
+ * Are `<missionFormValues>` complete enough to be transformed into a `MissionData` type and sent to the API?
+ */
+export function isCompleteMissionFormValues(missionFormValues: MissionFormValues | undefined): boolean {
+  try {
+    if (!missionFormValues) {
+      return false
+    }
+
+    getMissionDataFromMissionFormValues(missionFormValues)
+
+    return true
+  } catch (_) {
+    return false
+  }
+}
+
+export function isValidControlUnit(
+  formResourceUnit: MissionFormValues['controlUnits'][0]
+): formResourceUnit is Mission['controlUnits'][0] {
+  return !hasMissingOrUndefinedValues(['administration', 'contact', 'id', 'name', 'resources'], formResourceUnit)
+}
+
+export function getMissionDataFromMissionFormValues(missionFormValues: MissionFormValues): MissionData {
+  if (!missionFormValues.inputDateTimeRangeUtc) {
+    throw new FormError(missionFormValues, 'inputDateTimeRangeUtc')
+  }
+
+  const missionBaseValues = omit(['controlUnits', 'inputDateTimeRangeUtc'], missionFormValues)
+
+  const validControlUnits = missionFormValues.controlUnits.filter(isValidControlUnit)
+  if (validControlUnits.length !== missionFormValues.controlUnits.length) {
+    throw new FormError(missionFormValues, 'controlUnits')
+  }
+  const [inputStartDateTimeUtc, inputEndDateTimeUtc] = missionFormValues.inputDateTimeRangeUtc
+  const missionSource = MissionSource.MONITORFISH
```

Comment:
This would be set as `MONITORFISH` within the `MonitorEnv` backend, so don't need to send it as a payload to the API

Reply:
Okey, but we'll check if the type is part of the API input payload (because on the other hand, having a not used parameter on MonitorEnv API is a bad practice, and the Swagger doc of the API might document this behavior)

Reply:
Understood, I agree on that ^^.